### PR TITLE
Migrate base_import_async to 12.0

### PR DESCRIPTION
**Pull request by guewen**
* [x] needs #101

The first commits are shared with #101, relevant commits are the 2 last ones.

### CODE REVIEW

File: base_import_async/__manifest__.py [hunk 1]
```diff
@@ -22,5 +22,5 @@
     'qweb': [
         'static/src/xml/import.xml',
     ],
-    'installable': False,
+    'installable': True,
```
Comment (lmignon) [inline base_import_async/__manifest__.py:25]: `development_status` ????

**Review by lmignon — APPROVED** [review]
Thank you @guewen

File: base_import_async/__manifest__.py (hunk 1 above)
Comment (guewen) [reply base_import_async/__manifest__.py:25]: added

**Review by lmignon — COMMENTED** [review]
Strange, the import functionality seems to not work at all on runbot....

**Comment by guewen** [comment]
> Strange, the import functionality seems to not work at all on runbot....

Because of the CORS issue?

> An unknown CORS error occured. The error probably originates from a JavaScript file served from a > different origin. (Opening your browser console might give you a hint on the error.)

**Comment by guewen** [comment]
On my tests on runbot, the jobs are generated but there is this CORS error that pops when the notification appears. And in the browser console I get:

```
uncaught exception: an exception occurred in a caller-provided callback function
XML Parsing Error: syntax error
Location: http://3341401-102-8f5137.runbot3.odoo-community.org/longpolling/poll
Line Number 1, Column 1:
```

There is a problem with the longpolling:
selection_395

The expected response for longpolling would have been something like (copied from odoo's runbot):
```
{"jsonrpc": "2.0", "id": 403105148, "result": [{"id": 406, "channel": ["381328-12-0-360212-all", "res.partner", 3], "message": {"type": "activity_updated", "activity_created": true}}]}
```

**Comment by lmignon** [comment]
@guewen Yes the CORS issue is one problem but in my case I was not able to create the job...

**Comment by lmignon** [comment]
@sbidoul @sbejaoui Do you use this addon on 12.0?

**Comment by sbidoul** [comment]
@lmignon I've not tried it yet.

**Comment by GSLabIt** [comment]
@guewen did u try to import products with url image? it seems to be failing, while importing "normally" it works properly.

Schermata 2019-09-18 alle 19 00 43

@lmignon 
> @guewen Yes the CORS issue is one problem but in my case I was not able to create the job...

it seems that, at least importing excel, `options` are not passed correctly; both `encoding` and `separator` are `''`, so it fails opening writer. 

Beside that i'm getting CORS problem as well ..

**edit:**: I can confirm issue problem _1-character string_ is raised by `encoding` and `separator` when importing excel file format. It will not forward `options` correctly; I fixed it having `encoding` failing to `'utf-8'` and `delimiter` to `','`.

**Comment by GSLabIt** [comment]
@guewen any news on CORS issue?

**Comment by guewen** [comment]
@GSLabIt No, I didn't have the opportunity to work on this.

**Comment by guewen** [comment]
I found the origin of the error (which is unrelated to CORS, only appearing as such), fixed in my last commit. I think we are ready to merge!

**Comment by GSLabIt** [comment]
@guewen, great news. Did u try importing image from url from excel?

have a look at https://github.com/OCA/queue/pull/102#issuecomment-532776170

**Comment by guewen** [comment]
> Did u try importing image from url from excel?

Nope, forgot about this one :) (never tried this actually)

**Comment by guewen** [comment]
Actually, by looking at the implementation, I think that only CSV files are supported (which sort of make sense since files are split per line). I think we should hide the options and add a comment about this in the readme. (and if someone wants to implement excel file supports, it can be added later).

**Comment by GSLabIt** [comment]
@guewen what about fixing it by having `encoding` failing to `'utf-8'` and `delimiter` to `','`?

**Comment by guewen** [comment]
You are right, not sure how but excel files are read correctly. I pushed a new commit with this correction.

**Comment by GSLabIt** [comment]
> You are right, not sure how but excel files are read correctly. I pushed a new commit with this correction.

That's because you'll have data parsed by: 
https://github.com/OCA/queue/blob/b6de8170c85aeb69040ae3eecc097c29052b9e85/base_import_async/models/base_import_import.py#L47

So first all file is parsed, than splitted. If  have very big file, parsing could end in timeout.

**Comment by guewen** [comment]
Thanks @GSLabIt  :)

**Review by GSLabIt — APPROVED** [review]
LGTM

**Pull request merged**